index.js

```diff
@@ -3,26 +3,38 @@ const path = require('path');
 
 const pluginName = 'Chunks2JsonWebpackPlugin';
 
+const defaultOptions = {
+    // ignore files emitted by HMR by default
+    excludeFile: /\.hot-update\.js$/,
+    // group chunks by extension
+    chunkGroupName: filename => /\.([a-z0-9]+(\.map)?)(\?.*)?$/.exec(filename)[1],
+    outputDir: process.cwd(),
+    filename: 'build-manifest.json',
+    // generate contents to save to manifest file
+    objectToString: result => JSON.stringify(result),
+    publicPath: ''
+};
+
 class Chunks2JsonWebpackPlugin {
     constructor(options) {
-        this.options = options;
+        this.options = Object.assign({}, defaultOptions, options);
         this.result = {};
     }
     apply(compiler) {
         compiler.hooks.emit.tap(pluginName, compilation => {
-            compilation.chunks.forEach((chunk) => {
-                if (this.result[chunk.name] === undefined) {
+            this.result = {};
+            compilation.chunks.forEach(chunk => {
+                if (!this.result[chunk.name]) {
```

Comment:
Why so? The condition is executed as expected so this change won't affect anything but will be less readable IMHO. But I can make it as you said if you insist since you are boss here :)

Reply:
Is our coding standard, we always tend to use explicit comparison - I think it increases the readability and more clearly expresses the intention. :)